### PR TITLE
feat(rules): update reset rules modal and restrict button display

### PR DESCRIPTION
**Pull request by ccailly**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !33542

Rules reset button changed to danger style.
Original neutral modal replaced by danger modal.
Button displayed only if user is on root entity and has configuration modification rights.

image

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -2553,23 +2553,26 @@ public static function getSpecificValueToDisplay($field, $values, array $options
...
                                 static::getFormURL(),
                                 ['action' => 'uninstall'],
                                 _x('button', 'Uninstall'),
                                 ['id' => $ID],
                                 '',
                                 'class="btn btn-danger w-100"'
                             ),
+                            'content' => sprintf(
+                                __('By uninstalling the "%s" plugin you will lose all the data of the plugin.'),
+                                $plugin->getField('name')
```
Comment (cedric-anne) [inline src/Plugin.php:2574]: `content` will not be escaped anymore due to the usage of the `|raw` filter
```suggestion
                                __s('By uninstalling the "%s" plugin you will lose all the data of the plugin.'),
                                htmlspecialchars($plugin->getField('name'))
```

File: src/Glpi/Marketplace/View.php [hunk 2]
```diff
@@ -884,19 +884,22 @@ public static function getButtons(string $plugin_key = ""): string
...
                                </button>',
-                'uninstall_btn' => '<a href="#" class="btn btn-danger w-100 modify_plugin"
+                'confirm_btn' => '<a href="#" class="btn btn-danger w-100 modify_plugin"
                                        data-action="uninstall_plugin"
                                        data-bs-dismiss="modal">
                                        ' . _x("button", "Uninstall") . '
                                    </a>',
+                'content' => sprintf(
+                    __('By uninstalling the "%s" plugin you will lose all the data of the plugin.'),
+                    $plugin_inst->getField('name')
```
Comment (cedric-anne) [inline src/Glpi/Marketplace/View.php:901]: `content` will not be escaped anymore due to the usage of the `|raw` filter
```suggestion
                                __s('By uninstalling the "%s" plugin you will lose all the data of the plugin.'),
                                htmlspecialchars($plugin_inst->getField('name'))
```

**Review by AdrienClairembault — COMMENTED** [review]
I feel like the component except too much html rendered content as parameters, which is not ideal as it make it difficult to configure it.

Here are my suggestions to simplify it:
`open_btn` -> This value is displayed directly as "raw" content before the modal container.
Wouldn't it make more sense to display it manually before calling the component ?

`content` -> In the two implementation we use simple strings without HTML tags. I guess it could be a simple string variable ?

`confirm_btn` => I guess we have to keep this one as HTML for now. Ideally it should just be a simple label but with the design of GLPI the button is used as either a link or a form so we can't really change it :/

**Review by trasher — APPROVED** [review]

File: src/Glpi/Marketplace/View.php [hunk 3]
```diff
@@ -884,19 +884,25 @@ public static function getButtons(string $plugin_key = ""): string
...
-            $buttons .= TemplateRenderer::getInstance()->render('components/plugin_uninstall_modal.html.twig', [
-                'plugin_name' => $plugin_inst->getField('name'),
+            $uninstall_label = __s("Uninstall");
+            $buttons .= <<<HTML
+                <button data-bs-toggle="modal"
+                        data-bs-target="#uninstallModal{$plugin_inst->getField('directory')}"
+                        title="{$uninstall_label}">
+                    <i class="ti ti-folder-x"></i>
+                </button>
+            HTML;
```
Comment (cedric-anne) [inline src/Glpi/Marketplace/View.php:894]:
```suggestion
HTML;
```

File: src/Plugin.php [hunk 4]
```diff
@@ -2553,23 +2553,30 @@ public static function getSpecificValueToDisplay($field, $values, array $options
...
-                            'plugin_name' => $plugin->getField('name'),
+                        $uninstall_label = __s("Uninstall");
+                        $output .= <<<TWIG
+                            <a class="pointer"><span class="fas fa-fw fa-folder-minus fa-2x me-1"
+                                data-bs-toggle="modal"
+                                data-bs-target="#uninstallModal{$plugin->getField('directory')}"
+                                title="{$uninstall_label}">
+                                <span class="sr-only">{$uninstall_label}</span>
+                            </span></a>
+                        TWIG;
```
Comment (cedric-anne) [inline src/Plugin.php:2564]:
```suggestion
TWIG;
```

File: src/RuleCollection.php [hunk 5]
```diff
@@ -610,27 +611,28 @@ public function showListRules($target, $options = [])
...
+                    </button>
+
+                    {% set reset_btn %}
+                        <a class="btn btn-danger w-100" role="button" href="{{ rule_class|itemtype_search_path }}?reinit=true&subtype={{ rule_class }}">
+                            {{ reset_label }}
+                        </a>
+                    {% endset %}
+
+                    {{ include('components/danger_modal.html.twig', {
+                        'open_btn': open_btn,
```
Comment (AdrienClairembault) [inline src/RuleCollection.php:625]:
```suggestion
```

I think this param is no longer used.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**